### PR TITLE
Implemented DoS security module. Now queries and mutations have selectable nesting depth limit.

### PR DESCRIPTION
**Pull request by pjmsullivan**
Implemented Denial of Service attack security module enabling devs to set a maximum query nesting depth value, above which, queries will be rejected.

# Checklist

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

# Related Issue

Previously, Obsidian did not allow for instantiation of a GraphQL security strategy.  This feature easily allows for a developer-designated value to establish a maximum query depth limit to prevent a DoS attack of deeply-nested or recursive queries from overloading  the GraphQL runtime.  

# Solution

The solution was to provide a new DoSSecurity.ts module that allows developers to specify the maximum nested query depth upon initialization of the ObsidianRouter.  By default, the security feature is disabled with a value of 0.  If a value is specified, all GraphQL queries and mutations will be evaluated against the maximum query depth, and queries that exceed the maximum will not be routed to the GraphQL runtime, neutralizing a potential nested query DoS attack.

# Additional Info

This module was written in typescript.

### CODE REVIEW

File: src/destructure.js [hunk 1]
```diff
@@ -12,7 +12,7 @@
  *
  */
 // this function will destructure a query/mutation operation string into a query/mutation operation object
-export function destructureQueries(queryOperationStr) {
+export default function destructureQueries(queryOperationStr) {
```
Comment (cssim22) [inline src/destructure.js:15]: Agree with updated export default syntax

File: src/obsidian.ts [hunk 2]
```diff
@@ -65,7 +68,7 @@ export async function ObsidianRouter<T>({
   // set redis configurations
 
   if (policy || maxmemory) {
-    console.log('inside if');
+    // console.log('inside if');
```
Comment (cssim22) [inline src/obsidian.ts:71]: Remove this console log

**Review by cssim22 — APPROVED** [review]
All changes look great and are working as intended.

File: test_files/rhum_test_files/DoSSecurity_test.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { Rhum } from 'https://deno.land/x/rhum@v1.1.4/mod.ts';
+import queryDepthLimiter from '../../src/DoSSecurity.ts';
+import { test } from '../test_variables/DoSSecurity_variables.ts';
+
+
+Rhum.testPlan('DoSSecurity.ts', () => {
+  Rhum.testSuite('Query depth limit NOT exceeded tests', () => {
+    Rhum.testCase('Test query depth with allowable depth 2', () => {
+      const results = queryDepthLimiter(test.DEPTH_2_QUERY, 10);
```
Comment (cssim22) [inline test_files/rhum_test_files/DoSSecurity_test.ts:9]: should change depth test value to 2

**Review by cssim22 — CHANGES_REQUESTED** [review]
Create additional tests to confirm function of depth check with multiple queries/mutations at the same time.

**Review by cssim22 — APPROVED** [review]
All changes have been reviewed and tested.

**Review by justinwmckay — APPROVED** [review]
Love it.

**Comment by cssim22** [comment]
Security measures reviewed and approved

**Pull request merged**